File: Leetcode/Daily/06-2025/Java/16.java
// Problem: 2016. Maximum Difference Between Increasing Elements (Easy)
// Link: https://leetcode.com/problems/maximum-difference-between-increasing-elements

class Solution {
    public int maximumDifference(int[] nums) {
        int n = nums.length, maxx = 0, ans = -1;

        for (int i = n-1; i >= 0; i--) {
            if (maxx > nums[i]) ans = Math.max(ans, maxx - nums[i]);
            maxx = Math.max(maxx, nums[i]);
        }

        return ans;  
    }
}

File: Leetcode/Daily/07-2025/C++/16.cpp
// Problem: 3201. Find the Maximum Length of Valid Subsequence I (Medium)
// Link: https://leetcode.com/problems/find-the-maximum-length-of-valid-subsequence-i

class Solution {
public:
    int maximumLength(vector<int>& nums) {
        int n = nums.size();
        
        bool sign = nums[0] % 2;
        int odd = sign, even = !sign, alt = 1;
        
        for (int i = 1; i  < n; i++) {
            if (nums[i] % 2) odd++;
            else even++;

            if (nums[i] % 2 != sign) {
                alt++;
                sign = !sign;
            }
        }

        return max({odd, even, alt});
    }
};